llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -627,12 +627,10 @@ bool canNarrowUserTreeToS20(MachineRegisterInfo &MRI, InstrNode Start,
     }
     switch (Use.getOpcode()) {
     case TargetOpcode::G_TRUNC:
-      // Sanity check that we are not truncating into less than 20 bits and
-      // losing precision. If this happens this means we missed an extension
-      // from that small type back to S20 to feed into our ptr.add intrinsics.
-      assert(MRI.getType(Use.getOperand(0).getReg()).getScalarSizeInBits() >=
-             20);
-      [[fallthrough]];
+      // Check if the scalar size of the operand's type is at least 20 bits,
+      // this ensures that the G_TRUNC can be safely converted to a COPY.
+      if (MRI.getType(Use.getOperand(0).getReg()).getScalarSizeInBits() < 20)
+        return false;
```

Comment:
We have a warning now:
```
../llvm/lib/Target/AIE/AIECombinerHelper.cpp:634:5: warning: unannotated fall-through between switch labels [-Wimplicit-fallthrough]
    case TargetOpcode::G_PTR_ADD:
    ^
../llvm/lib/Target/AIE/AIECombinerHelper.cpp:634:5: note: insert 'LLVM_FALLTHROUGH;' to silence this warning
    case TargetOpcode::G_PTR_ADD:
    ^
    LLVM_FALLTHROUGH; 
../llvm/lib/Target/AIE/AIECombinerHelper.cpp:634:5: note: insert 'break;' to avoid fall-through
    case TargetOpcode::G_PTR_ADD:
    ^
    break; 
```

Reply:
Adding `[[fallthrough]];`